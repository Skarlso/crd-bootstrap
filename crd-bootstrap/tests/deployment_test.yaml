suite: test deployment
templates:
  - deployment_crd-bootstrap-controller-manager.yaml
tests:
  - it: deployment should match snapshot
    set:
      image.tag: v0.8.0
    asserts:
      - matchSnapshot: {}
