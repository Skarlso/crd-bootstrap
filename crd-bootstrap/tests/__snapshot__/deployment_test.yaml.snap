deployment should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: manager
        app.kubernetes.io/created-by: crd-bootstrap
        app.kubernetes.io/instance: controller-manager
        app.kubernetes.io/managed-by: kustomize
        app.kubernetes.io/name: deployment
        app.kubernetes.io/part-of: crd-bootstrap
        control-plane: crd-bootstrap-controller-manager
      name: crd-bootstrap-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          control-plane: crd-bootstrap-controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            control-plane: crd-bootstrap-controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
              command:
                - /manager
              image: ghcr.io/skarlso/crd-bootstrap-controller:v0.8.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
          securityContext:
            runAsNonRoot: true
          serviceAccountName: crd-bootstrap-controller-manager
          terminationGracePeriodSeconds: 10
